overlord/configstate/configcore/picfg.go

```diff
@@ -88,16 +89,35 @@ func updatePiConfig(path string, config map[string]string) error {
 	return nil
 }
 
-func piConfigFile(opts *fsOnlyContext) string {
+func piConfigFile(opts *fsOnlyContext) (string, error) {
 	rootDir := dirs.GlobalRootDir
+	subdir := "/boot/uboot"
 	if opts != nil {
 		rootDir = opts.RootDir
+	} else {
```

Comment:
it occurs to me now that this shouldn't be in this else, for example even in the initramfs on uc20 we want to be applying config changes to /run/mnt/ubutu-seed/config.txt and not to /run/mnt/data/writable/system-data/ this should be outside of this. I have this fixed in https://github.com/snapcore/snapd/pull/9965, but I suppose if folks want to I could fix it here instead, the git history has gotten a bit muddled unfortunately...

Reply:
yes, the original subdir value is always wrong for UC20